tests/integration/dynamic_ip_test.yaml

```diff
@@ -0,0 +1,1467 @@
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+  ipv6:
+    enabled: true
+    dhcp: true
+    autoconf: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: Remove the interface dhcpcli
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: absent
+description: Configure the ethernet device dhcpcli to have IPv6 dhcp-duid 0f:66:55:BC:73:4D configured
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv6:
+    dhcp-duid: 0f:66:55:BC:73:4D
+description: Configure the dhcpcli ethernet interface with DHCP4, IPv4 auto dns, IPv4 auto gateway, IPv4 auto routes, IPv4 auto route table ID 0, DHCP6, autoconf, IPv6 auto dns, IPv6 auto gateway, IPv6 auto routes, IPv6 auto route table ID 0
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+  ipv6:
+    enabled: true
+    dhcp: true
+    autoconf: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: Remove the interface dhcpcli
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: absent
+description: Configure the dummy00 interface with the DHCP4, IPv4 auto dns, IPv4 auto gateway, IPv4 auto routes, IPv4 auto route table ID 0, DHCP6, autoconf, IPv6 auto dns, IPv6 auto gateway, IPv6 auto routes, IPv6 auto route table ID 0
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+  ipv6:
+    enabled: true
+    dhcp: true
+    autoconf: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: Bring up the dummy device dummy00
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+description: Disable the IPv4 and IPv6 on the dummy00 interface
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+  ipv4:
+    enabled: false
+  ipv6:
+    enabled: false
+description: Delete the dummy device dummy00
+interfaces:
+- name: dummy00
+  type: dummy
+  state: absent
+description: Do not apply DNS resolver information and the routes retrieved from DHCP4 on the ethernet device eth0. Do not apply DNS resolver information and the routes retrieved from autoconf
+interfaces:
+- name: eth0
+  type: ethernet
+  state: up
+  ipv4:
+    auto-dns: false
+    auto-routes: false
+  ipv6:
+    auto-dns: false
+    auto-routes: false
+description: Configure the DNS server to 10.11.5.19 and 10.2.32.1, configure the ethernet device dhcpcli to have the IPv4 and IPv6 disabled. Configure the ethernet device eth0 to apply DNS resolver information and the routes retrieved from DHCP4. Configure the ethernet device eth0 to apply DNS resolver information and the routes retrieved from autoconf.
```

Comment:
`Set the static DNS nameservers to 10.11.5.19 and 10.2.32.1 and empty search domains`.